shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1679,37 +1755,45 @@ - (UITextPosition*)closestPositionToPoint:(CGPoint)point withinRange:(UITextRang
   NSUInteger start = ((FlutterTextPosition*)range.start).index;
   NSUInteger end = ((FlutterTextPosition*)range.end).index;
 
-  NSUInteger _closestIndex = 0;
-  CGRect _closestRect = CGRectZero;
-  NSUInteger _closestPosition = 0;
+  // Selecting text using the floating cursor is not as precise as the pencil.
+  // Allow further vertical deviation and base more of the decision on horizontal comparison.
+  CGFloat verticalPrecision = _isFloatingCursorActive ? 10 : 1;
+
+  BOOL isFirst = YES;
+  NSUInteger _closestRectIndex = 0;
   for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
```

Comment:
instead of using `isFirst` flag, can we just loop from `1` to `count`?

Reply:
`_selectionRects[i]` might not be within the `range`. We want `isFirst` only for the first rect within the range.